src/transport/SessionManager.h

```diff
@@ -206,7 +207,8 @@ class DLL_EXPORT SessionManager : public TransportMgrDelegate
     Optional<SessionHandle> CreateUnauthenticatedSession(const Transport::PeerAddress & peerAddress,
                                                          const ReliableMessageProtocolConfig & config)
     {
-        return mUnauthenticatedSessions.FindOrAllocateEntry(peerAddress, config);
+        NodeId ephemeralInitiatorNodeID = static_cast<NodeId>(Crypto::GetRandU64());
```

Comment:
If you consider these "node IDs" in that context, and in `GetPeerNodeId` you have;

```
 NodeId GetPeerNodeId() const
    {
        if (mSessionRole == SessionRole::kInitiator)
        {
            return kUndefinedNodeId;
        }
```

Then should we not avoid the value 0 here?

Reply:
Actually, the spec is slightly more restrictive than that even.  The operational node ID range was designated for this.

Reply:
Should I allocate the ephemeral node ID from Temporary Local Node ID range or Operational Node ID range

Reply:
No, it has to be allocated from the Operational Node ID range.

This is because we are relying on this (usually) not colliding.  The bigger operational node ID range is therefore better, and this is how the spec is currently written per https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/4796.

I view this as a reasonable use of the operational node ID space because these are effectively no-fabric node IDs, so don't collide with operational node IDs that exist on a fabric.

Reply:
The updated code still isn't quite right.

See https://github.com/project-chip/connectedhomeip/pull/14382#discussion_r794717551.